spacepy/coordinates.py

```diff
@@ -391,3 +397,78 @@ def append(self, other):
         data.extend(list(otherdata.data))
         newobj = Coords(data, dtype=self.dtype, carsph=self.carsph)
         return newobj
+
+    # -----------------------------------------------
+    def to_skycoord(self):
+        '''
+        Create an Astropy SkyCoord instance based on this instance
+
+        Returns
+        =======
+        out : Astropy SkyCoord instance
+            This coordinate as an Astropy SkyCoord
+
+        Notes
+        =====
+        This method requires Astropy to be installed.
+
+        This method uses the GEO coordinate frame as the common frame between the two libraries.
+        '''
+        if self.ticks is None:
+            raise ValueError("This method requires the attribute `ticks` to be specified.")
+
+        try:
+            from astropy.coordinates import SkyCoord
+            from astropy.time import Time
+        except ImportError as e:
+            raise e.__class__("This method requires Astropy to be installed.")
+
+        # Convert to GEO (i.e., Astropy's ITRS)
```

Comment:
This style of comment, statement, blank line soaks up vertical space and isn't adding much...it's obvoius it's converting to GEO, just need the ITRS info. Since the github multi-line suggestion thing doesn't work well, I'm envisioning something like:

```
coord = self.convert('GEO', 'car') # Astropy ITRS is GEO
data = coord.data * self.Re # metres
obstime = astropy.time.Time(coord.ticks.ISO, format='isot', scale='utc')
```

As @drsteve suggested, maybe use, say, TAI instead of ISO ticks?

Reply:
There's a _big_ nomenclature difference between TAI as used in key sections of the space physics community, and TAI as used by the broader astronomical and astrodynamics community (which is the correct one, of course). We use the "incorrect" version because it's fairly well entrenched in a number of key areas of magnetospheric physics. Our "TAI" is integer seconds since 1958/1/1, where the _proper_ TAI is offset from UTC by the number of leapseconds.
For a numeric time system, I'd probably use MJD. Although I think UTC datetimes get generated for pretty much any input time system, so it's probably least overhead to use those rather than forcing a conversion. If a conversion is desired, then MJD is my preference.
ISO strings are fine by me as long as we don't think the performance hit will be an issue.

TL;DR - MJD or UTC datetimes are my preference. ISO is fine if the performance hit is negligible.

Reply:
So...  It turns out that how you handle ISO, MJD, and UTC are all problematic because I believe they all repeat at the leap second.  After some experimentation, I believe the most sensible approach is to define a custom format for Astropy Time that matches your definition for "TAI".  That way, there is no degeneracy at the leap second and conversion overhead is kept to a minimum.

This custom time format is defined at the bottom of `spacepy.coordinates` to keep it conceptually grouped with the SkyCoord conversion functions, but of course it may make sense to move it to `spacepy.time`.

Example:
```python
>>> from spacepy.time import Ticktock
>>> Ticktock(1861920037.0, 'TAI').ISO
dmarray(['2017-01-01T00:00:00'], dtype='<U19')

>>> import spacepy.coordinates
>>> import astropy.time
>>> Time('2017-01-01 00:00:00', scale='utc').spacepy_tai
1861920037.0
>>> Time(1861920037.0, format='spacepy_tai').utc.iso
'2017-01-01 00:00:00.000'
```